### ISSUE TITLE
Eclipse crashes when opening GWT Designer visual editor with WebKitGTK 1.10

### ISSUE THREAD

**Issue by GoogleCodeExporter** [clayberg/gwt-designer#4]
```
What steps will reproduce the problem?
1. Open an appropriate file with the GWT Designer editor and switch to design 
tab

What is the expected output? What do you see instead?

I expect to see the visual designer, instead Eclipse either hangs on a modal 
progress dialog saying "Initializing GWT Development Mode" indefinitely, or 
crashes with the following output:

#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fe14ddca5a0, pid=27698, tid=140606793508608
#
# JRE version: 7.0_21-b11
# Java VM: Java HotSpot(TM) 64-Bit Server VM (23.21-b01 mixed mode linux-amd64 
compressed oops)
# Problematic frame:
# C  [libwebkitgtk-1.0.so.0+0x14245a0]  void 
WTF::freeOwnedGPtr<_GdkEvent>(_GdkEvent*)+0x15df0
#
# Failed to write core dump. Core dumps have been disabled. To enable core 
dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/lorenzo/opt/hs_err_pid27698.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.sun.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

Please find the error report attached (hs_err_pid16600.log).

What version of the product are you using? On what operating system?

- Eclipse Juno SR2
- GWT Designer 3.1.2.r42x201303061525 (from the Google Plugin update site at 
http://dl.google.com/eclipse/plugin/4.2)
- Ubuntu 12.10 64 bit
- libwebkitgtk-1.0-0 version 1.10.0-0ubuntu1.1

Please provide any additional information below.

The issue is due to a bug in WebKitGTK 1.10 (which is the package repository 
version on Ubuntu 12.10). The SWT browser widget has the same problem, see 
https://bugs.eclipse.org/bugs/show_bug.cgi?id=392967

I've attached a patch that fixes the issue, and published a feature patch that 
can be installed as a workaround until a fixed version of gwt designer will 
come out. You can find the feature patch at this update site:

http://lorecarra.appspot.com/update-sites/gwt-designer-patches/


```

Original issue reported on code.google.com by `lorenzo....@cubica.eu` on 23 Apr 2013 at 11:03

Attachments:
- [bs-webkit-window-size.patch](https://storage.googleapis.com/google-code-attachments/gwt-designer/issue-4/comment-0/bs-webkit-window-size.patch)
- [hs_err_pid16600.log](https://storage.googleapis.com/google-code-attachments/gwt-designer/issue-4/comment-0/hs_err_pid16600.log)

**Comment by GoogleCodeExporter** [comment]
```
I've updated the update site 
(http://lorecarra.appspot.com/update-sites/gwt-designer-patches/), now it 
contains 2 feature patches: 

- one is for the GPE version of GWT Designer, installable from the GPE update 
site (http://dl.google.com/eclipse/plugin/4.2)
- the other is for the complete (but older) GWT Designer, installable from the 
GWT Designer update site (http://dl.google.com/eclipse/inst/d2gwt/latest/4.2)

In order to install the GWT Designer version, uncheck the "Show only the latest 
versions of available software", and choose ONLY the "GWT Designer Version" 
feature patch. Otherwise, if you want to install the GPE version, only install 
that one.
```

Original comment by `lorenzo....@cubica.eu` on 24 Apr 2013 at 7:33

**Comment by GoogleCodeExporter** [comment]
```
Thanks a lot! This fix works great for Eclipse 4.2.2! It did not work for my 
issue 
(https://code.google.com/p/google-plugin-for-eclipse/issues/detail?id=224) with 
Eclipse 4.3 though.
```

Original comment by `Splak...@gmail.com` on 14 Jul 2013 at 9:34

**Comment by GoogleCodeExporter** [comment]
```
It would be great to get this patch into the GPE, WindowBuilder, or GWT, which 
one does it actually need to go into?
```

Original comment by `Splak...@gmail.com` on 14 Jul 2013 at 9:35

**Comment by GoogleCodeExporter** [comment]
```
From Eric Clayberg: "The issue and the patch you reference appear to be for GWT 
Designer rather than WindowBuilder. In fact, it looks like a patch to the code 
that GWTD used to interact with WebKit."

So it looks like this issue needs to be fixed in GWTD or GPE and not as part of 
the Eclipse WindowBuilder or GWT projects.
```

Original comment by `Splak...@gmail.com` on 14 Jul 2013 at 11:25

**Comment by GoogleCodeExporter** [comment]
```
@Lorenzo, thanks for the patch. I completed it to support Test/Preview feature 
and merged into trunk.
```

Original comment by `alexande...@gmail.com` on 15 Jul 2013 at 4:06

**Comment by GoogleCodeExporter** [comment]
```
New GWT Designer builds which include this patch are available from the 
following update sites:

http://dl.google.com/eclipse/inst/d2gwt/latest/3.6
http://dl.google.com/eclipse/inst/d2gwt/latest/3.7
http://dl.google.com/eclipse/inst/d2gwt/latest/4.2
http://dl.google.com/eclipse/inst/d2gwt/latest/4.3

Use the 4.2 link for 3.8
```

Original comment by `clayberg@google.com` on 18 Jul 2013 at 2:13

**Comment by GoogleCodeExporter** [comment]
```
Note that these are update site URLs for use with the Eclipse update manager. 
They are not direct download links.
```

Original comment by `clayberg@google.com` on 18 Jul 2013 at 2:14

**Comment by GoogleCodeExporter** [comment]
```
I was having this problem with GWT Designer found on "Google Update Site for 
Eclipse 4.3 - http://dl.google.com/eclipse/plugin/4.3", yesterday evening.  
This morning I updated with the repo Clay mentions in comment 6 and can confirm 
it works.

Is someone going to update the plugin on the main repo?  So others don't have 
to track this down.

Thanks for all the great work guys.
```

Original comment by `markfi...@initiativeassets.com` on 26 Aug 2013 at 1:59
